components/core/src/clp/regex_utils/regex_translation_utils.cpp

```diff
@@ -114,6 +152,36 @@ using StateTransitionFuncSig
  */
 [[nodiscard]] StateTransitionFuncSig final_state_cleanup;
 
+// Other helpers
+/**
+ * Appends a single character as a literal to the wildcard string.
+ *
+ * If the literal is a metacharacter in the wildcard syntax, prepend the literal with an escape
+ * backslash.
+ * @param ch The literal to be appended.
+ * @param wildcard_str The wildcard string to be updated.
+ */
+inline auto append_char_to_wildcard(char const ch, string& wildcard_str) -> void {
```

Comment:
```suggestion
auto append_char_to_wildcard(char const ch, string& wildcard_str) -> void {
```
1. We don't explicitly add `inline` unless the profiling shows a different on performance
2. We should put helper declarations at the top and move the actual implementation to the bottom, inside the anon namespace.